src/main/scala/com/ing/wbaa/gargoyle/proxy/api/ProxyService.scala

```diff
@@ -0,0 +1,88 @@
+package com.ing.wbaa.gargoyle.proxy.api
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.{ HttpRequest, HttpResponse, RemoteAddress, StatusCodes }
+import akka.http.scaladsl.server.Route
+import com.ing.wbaa.gargoyle.proxy.api.directive.ProxyDirectives
+import com.ing.wbaa.gargoyle.proxy.data.{ AwsAccessKey, AwsRequestCredential, S3Request, User }
+import com.typesafe.scalalogging.LazyLogging
+
+import scala.concurrent.Future
+import scala.util.{ Failure, Success }
+
+trait ProxyService extends LazyLogging {
+
+  // no validation of request currently
+  // once we get comfortable with get/put/del we can add permCheck
+  import ProxyDirectives._
+  import akka.http.scaladsl.server.Directives._
+
+  implicit def system: ActorSystem
+
+  // Request Handler methods
+  def validateUserRequest(request: HttpRequest, secretKey: String): Boolean
+  def executeRequest(request: HttpRequest, clientAddress: RemoteAddress): Future[HttpResponse]
+
+  // Authentication methods
+  def getUser(accessKey: AwsAccessKey): Future[Option[User]]
+  def isAuthenticated(awsRequestCredential: AwsRequestCredential): Future[Boolean]
+
+  // Authorization methods
+  def isAuthorized(request: S3Request, user: User): Boolean
+
+  val proxyServiceRoute: Route =
```

Comment:
too long val :)

Reply:
I agree, we should be able to split these up in separate directives I presume. Not sure yet how to do this properly though. If you know how let me know ;)